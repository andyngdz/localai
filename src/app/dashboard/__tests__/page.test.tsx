import { render, screen } from "@testing-library/react";
import { describe, expect, it, vi } from "vitest";
import DashboardScreen from "../page";

// Mock the Dashboard component
vi.mock("@/features/dashboard/presentations/Dashboard", () => ({
  Dashboard: () => <div data-testid="dashboard">Dashboard Component</div>,
}));

describe("DashboardScreen", () => {
  it("renders Dashboard component", () => {
    render(<DashboardScreen />);

    expect(screen.getByTestId("dashboard")).toBeInTheDocument();
    expect(screen.getByText("Dashboard Component")).toBeInTheDocument();
  });

  it("returns Dashboard as default export", () => {
    const { container } = render(<DashboardScreen />);

    expect(container.firstChild).toBeInTheDocument();
    expect(screen.getByTestId("dashboard")).toBeInTheDocument();
  });

  it("has correct component structure", () => {
    const { container } = render(<DashboardScreen />);

    // Should render only the Dashboard component
    expect(container.children).toHaveLength(1);
    expect(screen.getByTestId("dashboard")).toBeInTheDocument();
  });

  it("matches snapshot", () => {
    const { container } = render(<DashboardScreen />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
