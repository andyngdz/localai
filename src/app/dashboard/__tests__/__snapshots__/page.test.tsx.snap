// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DashboardScreen > matches snapshot 1`] = `
<div
  data-testid="dashboard"
>
  Dashboard Component
</div>
`;
